src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -8217,6 +8229,16 @@ namespace VariableSpeedCoils {
         return RatedSourceTemp;
     }
 
+    bool setVarSpeedCoilOperationFlag(Real64 const userSuppliedTotCoolCap)
```

Comment:
A function seems a bit overkill here, since this could simply be
```
state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).coilOperationFlag = (state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).coilOperationFlag > 0.0);
```

Reply:
I think you mean RatedCapCoolTotal > 0

Reply:
Oh, but the coil could be autosized. And the coil could autosize to 0 if there were no load. So this should be set after sizing. Maybe a few selectively placed If (RatedCapCoolTotal > 0) conditionals could get you past the divide by 0 issues and allow the sim to proceed. In the Calc function are several if (something) SimFlag = false/return. Maybe add another for if (RatedCapCoolTotal == 0.0) SimFlag = false/return and not need a new bool? or tag this to one of those check:

        if (CompressorOp == CompressorOperation::Off || RatedCapCoolTotal == 0.0) {
            state.dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).SimFlag = false;
            return;
        }

Reply:
@rraustad Given your knowledge of the Variable Speed Coil work and looking at your suggestion, I will try this.  Since the issue is on the cooling side, it seems likely that I won't have to do what you are suggesting in a bunch of places.  So, my plan is to back out what I've done so far, implement/test the type of fix you are suggesting above, and then either move forward with that or report back here (if this didn't work).  Third time is the charm?

Reply:
Ok, so the adding of the condition in the Calc routine did not avoid the infinite loop.  I traced this back further to the the Init routine which called the Size routine which called CalcCBF where things still got stuck.

As a result, I backed out the added conditions and skipped the Size routine in the Init step (around line 3990) when RatedCapCoolTotal was zero.  This avoided the infinite loop...but the air side coil inlet and outlet conditions were not matching.

So, I put the added conditions suggested by @rraustad back in and kept the new skip protocol described in the previous paragraph.  Now, we are back to where we should be with the same results as with my "Take2" for three different cases. In other words, when I run both Take2 and the latest iteration described here both for a zero value of RatedCapCoolTotal, I get identical results (.csv show no diffs).  I repeated that for AutoSize and a fixed value for RaterCapCoolTotal.  No diffs between Take2 and the latest iteration for each of those variations.

I've now pushed that commit up.  It backs out everything done previously in "Take2" (except the documentation change which stays) and puts the new changes in.  There are three changes to what has been done: 1. the skipping of the Size routine called from the Init around line 3990 and 2./3. the addition of the change suggested by @rraustad for the SimFlag = false/return statements (two places). 

@mjwitte @rraustad Please look this over and see if it is more to your liking.  I will start figuring out a unit test.